### PR TITLE
Clarify the distinction between line breaking and text segmentation

### LINKED ISSUE #3261: Improve documentation of Segmenter
There are a number of questions that have been raised, such as https://github.com/unicode-org/icu4x/issues/3236, and why some segmenters return a breakpoint at index 0 but others not. We should provide 1-2 paragraphs of explanatory text for each of the 4 segmenters, with a link to the spec for more details.

### PR DESCRIPTION
**Pull request by eggrobin**
In particular:
- avoid the term _break opportunity_ in the context of the UAX #&zwnj;29 algorithms;
- for each of the UAX #&zwnj;29 algorithms, provide an example that explicitly computes the segments;
- for the LineSegmenter, clarify that both mandatory breaks and break opportunities are returned, and that these do not determine segments.

I think this fixes #3261 as far as the technical content is concerned; but those explanations of the four segmenters are perhaps weirdly buried in the examples. Editorial suggestions of a better structure are welcome.

### CODE REVIEW

File: experimental/segmenter/src/grapheme.rs
Comment (Manishearth) [inline experimental/segmenter/src/grapheme.rs:76]: praise: :cat:

File: experimental/segmenter/src/line.rs
Comment (Manishearth) [inline experimental/segmenter/src/line.rs:165]: suggestion: _may_ be worth saying that it is possible for callers to distinguish post-facto by checking the characters against a very small list, since learning that requires some poking around the spec.

**Review by Manishearth — CHANGES_REQUESTED** [review]
lgtm, may want approval from someone else too

**Review by sffc — COMMENTED** [review]
Thank you! It reads clearly; I'll defer to @aethanyc and @Manishearth on the content

File: experimental/segmenter/src/line.rs
Comment (eggrobin) [reply experimental/segmenter/src/line.rs]: Done, with an example.

File: experimental/segmenter/src/line.rs
Comment (Manishearth) [inline experimental/segmenter/src/line.rs:193]: nit: To avoid people copying this entire code I would likely make this a separate example, contextualized with "if you need to know the mandatory breaks here's what you can do".

Note that in Rust doctests you can hide setup lines by using `#`

**Review by Manishearth — APPROVED** [review]

File: experimental/segmenter/src/line.rs
Comment (aethanyc) [inline experimental/segmenter/src/line.rs:179]: thought (non-blocking): since the example has been moved to the bottom of the example section, it's probably worth moving to line 140, after the paragraph "Supports loading line break data....", if we feel this is an useful explanation that every line segmenter user should be aware of.

**Review by aethanyc — APPROVED** [review]
praise: I like the example demonstrating how to get segments from boundaries.

**Review by aethanyc — APPROVED** [review]

**Pull request merged**
